parachains/pallets/bridge-assets-transfer/src/lib.rs

```diff
@@ -67,6 +69,28 @@ pub mod pallet {
 	#[pallet::generate_store(pub(super) trait Store)]
 	pub struct Pallet<T>(_);
 
+	/// Everything we need to run benchmarks.
+	#[cfg(feature = "runtime-benchmarks")]
+	pub trait BenchmarkHelper<RuntimeOrigin> {
+		/// Returns proper bridge configuration, supported by the runtime.
+		///
+		/// We expect that the XCM environment (`BridgeXcmSender`) has everything enabled
+		/// to support transfer to this destination **after** we our `prepare_transfer` call.
+		fn bridge_config() -> (NetworkId, BridgeConfig);
+		/// Prepare environment for assets transfer and return transfer origin and assets
+		/// to transfer. After this function is called, we expect `transfer_asset_via_bridge`
+		/// to succeed, so in proper environment, it should:
+		///
+		/// - deposit enough funds (fee from `bridge_config()` and transferred assets) to the sender account;
+		///
+		/// - ensure that the `BridgeXcmSender` is properly configured for the transfer;
+		///
+		/// - be close to the worst possible scenario - i.e. if some account may need to be created during
```

Comment:
I've also been thinking about supporting different bridges here - e.g. in our case message will be sent to bridge hub first. But what if in other case message needs to be sent using some other queueing mechanism? E.g. when it is deployed at the same chain where `pallet-bridge-messages` is deployed or where eth bridge is deployed? Right now the assumption is that we won't do any refunds + the worst case needs to be "prepared" here. If we need to support such ^^^ case, then we may need to change sending interface a bit (to return actual used weight)

Reply:
> as we expect eth bridge deployed on bridge-hub (now).

Sorry - my comment is ambiguous. It was about the case when it works in a different way :) I mean - in our case it doesn't matter, but since anyone may use this pallet at their own parachain and use in a different way, we may think about it in advance. But let's just stay with our own case for now

> @svyatonik Slava, what do you mean by assumption is that we won't do any refunds? Does it mean, that we dont do any withdrawal for BuyExecution instruction here?

No - I mean that if there are several bridges and using first is `1_000_000` weight units and second is `5_000_000`, then we could return worst case from there and then return actual used weight from the call (i.e. "refund" the excessive cost of the call). But again - for our case it isn't an issue, since we're going to use the same transport everywhere

Reply:
TLDR: nvm for now - it is irrelevant for our planned deployment

Reply:
> Sorry - my comment is ambiguous. It was about the case when it works in a different way :) I mean - in our case it doesn't matter, but since anyone may use this pallet at their own parachain and use in a different way, we may think about it in advance. But let's just stay with our own case for now

yes, I was also thinking about different usages of this pallet, with `type BridgeXcmSender: SendXcm` you can send it everywhere (to sibling, to relay, to some local pallets..).
what would you suggest here to better support "use in a different way"?

Reply:
Something like having a dedicated trait that returns actual/unspent weight when XCM is sent (and of course is using some `SendXcm` implementation inside). So the weight could be different for every possible route (HRMP/local pallet call). Then we could return post-dispatch weight from the call. But yeah - let's not do this now :)